packages/locale/src/format/cldr/CldrZoneTextPrinterParser.js

```diff
@@ -25,6 +24,18 @@ const LENGTH_COMPARATOR = (str1, str2) => {
     }
     return cmp;
 };
+
+/**
+ * Cache for `_cachedResolveZoneIdText`.
+ * 
+ * Its basic structure is:
+ * Map { cldrInstance: zoneId }
+ * Obj { zoneId: style}
+ * Obj { style: type}
+ * Obj { type: resolvedZoneIdText}
+ */
+const resolveZoneIdTextCache = new Map();
```

Comment:
Technically this cache could get really big if you have a large number of `cldr locales x zone ids to print x styles x types`, but that seems unlikely in most use cases. Without this cache, our 1000 iterations take about 90ms, with this cache, things take the 7ms in the PR summary.